middleware/validateRegister.js

```diff
@@ -0,0 +1,35 @@
+import Joi from 'joi';
+
+export const validateRegister = (req, res, next) => {
+  const schema = Joi.object({
+    email: Joi.string()
+      .email({ tlds: { allow: false } })
+      .required()
+      .messages({
+        'string.email': 'Invalid email format.',
+        'any.required': 'Email is required.',
+      }),
+    name: Joi.string().min(2).required().messages({
+      'string.min': 'Name must be at least 2 characters long.',
+      'any.required': 'Name is required.',
+    }),
+    password: Joi.string()
+      .min(8)
+      .pattern(new RegExp('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*])'))
+      .required()
+      .messages({
+        'string.min': 'Password must be at least 8 characters long.',
+        'string.pattern.base':
+          'Password must contain at least one uppercase letter, one lowercase letter, one number, and one special character.',
+        'any.required': 'Password is required.',
+      }),
+  });
+
+  const { error } = schema.validate(req.body);
+
+  if (error) {
+    return res.status(400).json({ message: error.details[0].message });
```

Comment:
On lecture, we were considering the approach with Error object and inherited classes for the custom errors, like validation error.  
Please check the example here: [https://github.com/VitalyaIlin/hillel-code-examples/tree/main/express-middleware](https://github.com/VitalyaIlin/hillel-code-examples/tree/main/express-middleware)